### PR TITLE
Parity with c# - binary compat changes for LuisRecognizer

### PR DESCRIPTION
**Pull request by daveta**
Matching changes to [C# PR for binary compat with LUIS recognizer](https://github.com/Microsoft/botbuilder-dotnet/pull/1561) which moves TelemetryClient into options class.

Also fix broken test and make hook async.

### CODE REVIEW

File: libraries/botbuilder-ai/tests/luisRecognizer.test.js [hunk 1]
```diff
@@ -115,7 +115,7 @@ function TestJson(file, done, includeAllIntents, includeInstance, telemetryClien
     var newPath = expectedPath + ".new";
     var context = new TestContext({ text: expected.text });
     var recognizer = new LuisRecognizer({ applicationId: luisAppId, endpointKey: endpointKey }, 
-        { includeAllIntents: includeAllIntents, includeInstanceData: includeInstance }, true, telemetryClient, logPersonalInformation);
+        { includeAllIntents: includeAllIntents, includeInstanceData: includeInstance, telemetryClient: telemetryClient, logPersonalInformation: logPersonalInformation }, true, telemetryClient, logPersonalInformation);
```
Comment (Zerryth) [inline libraries/botbuilder-ai/tests/luisRecognizer.test.js]: It looks like you moved telemetry & PII into the options argument, which is great!
But shouldn't we also remove the extra `telemetryClient` and `logPersonalInformation` that comes after "`true`" (the `includeApiResults` parameter)?

File: libraries/botbuilder-ai/src/luisRecognizer.ts [hunk 2]
```diff
@@ -316,7 +327,7 @@ export class LuisRecognizer implements LuisRecognizerTelemetryClient{
      * @param telemetryProperties Additional properties to be logged to telemetry with the LuisResult event.
      * @param telemetryMetrics Additional metrics to be logged to telemetry with the LuisResult event.
      */
-    protected onRecognizerResults(recognizerResult: RecognizerResult, turnContext: TurnContext, telemetryProperties?: {[key: string]:string}, telemetryMetrics?: {[key: string]:number}): Promise<void> {
+    protected async onRecognizerResults(recognizerResult: RecognizerResult, turnContext: TurnContext, telemetryProperties?: {[key: string]:string}, telemetryMetrics?: {[key: string]:number}): Promise<void> {
         this.fillTelemetryProperties(recognizerResult, turnContext, telemetryProperties).then(props => {
```
Comment (Zerryth) [inline libraries/botbuilder-ai/src/luisRecognizer.ts]: think we need an `await` for `this.fillTelemetryProperties()`

Comment (Zerryth) [reply libraries/botbuilder-ai/src/luisRecognizer.ts]: Looks like you did await on the C# side too 
https://github.com/Microsoft/botbuilder-dotnet/blob/5b3075cbac8a2c15d953282f472ced63dd6f931b/libraries/Microsoft.Bot.Builder.AI.LUIS/LuisRecognizer.cs#L190

**Comment by Zerryth** [comment]
Looks parity-fulfilling 👍 
Just a couple things to address, but other than that looks good

File: libraries/botbuilder-ai/tests/luisRecognizer.test.js (hunk 1 above)
Comment (daveta) [reply libraries/botbuilder-ai/tests/luisRecognizer.test.js]: Awesome!  Thanks for the feedback!

File: libraries/botbuilder-ai/src/luisRecognizer.ts (hunk 2 above)
Comment (daveta) [reply libraries/botbuilder-ai/src/luisRecognizer.ts]: Great catch!!

**Pull request merged**